File: backend/app/domain/transactions/status.py
"""Transaction status predicates — port of src/domain/transactions/status.ts."""


def is_in_flight_status(status: str | None = None) -> bool:
    return status in ("queued", "processing", "merging")


def is_failed_status(status: str | None = None) -> bool:
    return status in ("failed", "merge_failed")


def is_merge_status(status: str | None = None) -> bool:
    return status in ("merging", "merge_failed")

File: frontend/src/domain/transactions/status.ts
import type { TransactionStatus } from "@/types";

export function isInFlightStatus(status?: TransactionStatus | null): boolean {
  return status === "queued" || status === "processing" || status === "merging";
}

export function isFailedStatus(status?: TransactionStatus | null): boolean {
  return status === "failed" || status === "merge_failed";
}

export function isMergeStatus(status?: TransactionStatus | null): boolean {
  return status === "merging" || status === "merge_failed";
}
